frontend/packages/console-dynamic-plugin-sdk/src/extensions/catalog.ts

```diff
@@ -6,19 +6,21 @@ namespace ExtensionProperties {
   export type CatalogItemType = {
     /** Type for the catalog item. */
     type: string;
-    /** Title fpr the catalog item. */
+    /** Title for the catalog item. */
     title: string;
     /** Description for the type specific catalog. */
-    catalogDescription: string;
+    catalogDescription?: string;
     /** Description for the catalog item type. */
-    typeDescription: string;
+    typeDescription?: string;
     /** Custom filters specific to the catalog item.  */
     filters?: CatalogItemAttribute[];
     /** Custom groupings specific to the catalog item. */
     groupings?: CatalogItemAttribute[];
   };
 
   export type CatalogItemProvider = {
+    /** The unique identifier for the catalog this provider contributes to. */
+    catalogId: string;
```

Comment:
Should this also be?
```suggestion
    catalogId: string | string[];
```
This way one provider can provide to multiple catalogs. For example we have `dev-catalog` and `topology-quick-add-catalog` getting almost same catalog items but quick starts is only supposed to be in topology quick add. There might be other items exclusive for quick add in future.

Reply:
Actually I'll make it just an array to avoid the different types.

Reply:
nm... i made your suggestion

Reply:
I actually prefer `string[]` myself, makes it simpler for the extension consumer. My suggestion was based on our earlier extensions where we use `string | string[]` for `insertBefore` and `insertAfter`.